### PR TITLE
Fix DiagnosticAnalyzerFieldsAnalyzer for delegate fields

### PR DESCRIPTION
**Pull request by CollinAlpert**
This PR removes the warning for storing per-compilation data inside an analyzer field, when that field is a delegate type.

### CODE REVIEW

**Comment by mavasani** [comment]
@CollinAlpert Is there a github issue tracking this bug or was this identified by you through dogfooding?

File: src/Microsoft.CodeAnalysis.Analyzers/VisualBasic/MetaAnalyzers/BasicDiagnosticAnalyzerFieldsAnalyzer.vb [hunk 1]
```diff
@@ -8,6 +9,25 @@ Namespace Microsoft.CodeAnalysis.VisualBasic.Analyzers.MetaAnalyzers
     <DiagnosticAnalyzer(LanguageNames.VisualBasic)>
     Public Class BasicDiagnosticAnalyzerFieldsAnalyzer
         Inherits DiagnosticAnalyzerFieldsAnalyzer(Of ClassBlockSyntax, StructureBlockSyntax, FieldDeclarationSyntax, TypeSyntax, SimpleAsClauseSyntax)
+
+        Protected Overrides Function IsContainedInFuncOrAction(typeSyntax As TypeSyntax, model As SemanticModel, funcs As ImmutableArray(Of INamedTypeSymbol), actions As ImmutableArray(Of INamedTypeSymbol)) As Boolean
```
Comment (mavasani) [inline src/Microsoft.CodeAnalysis.Analyzers/VisualBasic/MetaAnalyzers/BasicDiagnosticAnalyzerFieldsAnalyzer.vb:13]: Can we added VB tests for this scenario?

**Comment by CollinAlpert** [comment]
@mavasani I noticed it while working on #6727, specifically [here](https://github.com/dotnet/roslyn-analyzers/blob/925976a672b2dd63d1314b6644332a6a327bcf73/src/NetAnalyzers/Core/Microsoft.NetCore.Analyzers/Runtime/UseStringEqualsOverStringCompare.cs#L295).

**Review by mavasani — APPROVED** [review]
LGTM. I can merge once you add VB tests. Thanks!

File: src/Microsoft.CodeAnalysis.Analyzers/VisualBasic/MetaAnalyzers/BasicDiagnosticAnalyzerFieldsAnalyzer.vb (hunk 1 above)
Comment (CollinAlpert) [reply src/Microsoft.CodeAnalysis.Analyzers/VisualBasic/MetaAnalyzers/BasicDiagnosticAnalyzerFieldsAnalyzer.vb]: Done. I simplified the check to use [`DelegateInvokeMethod`](https://learn.microsoft.com/en-us/dotnet/api/microsoft.codeanalysis.inamedtypesymbol.delegateinvokemethod?view=roslyn-dotnet-4.6.0#microsoft-codeanalysis-inamedtypesymbol-delegateinvokemethod).

**Review by mavasani — APPROVED** [review]
Thanks!

**Pull request merged**